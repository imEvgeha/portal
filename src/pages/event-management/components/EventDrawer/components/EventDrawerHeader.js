import React from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {connect} from 'react-redux';
import {createLoadingSelector} from '../../../../../ui/loading/loadingSelectors';
import {REPLAY_EVENT} from '../../../eventManagementActionTypes';
import {replayEvent} from '../../../eventManagementActions';

const EventDrawerHeader = ({event, isReplaying, onReplay}) => {
    const onInnerReplay = () => {
        const payload = {eventId: event.eventId};
        onReplay(payload);
    };

    return (
        <div className='nexus-c-event-drawer-header'>
            <Button onClick={onInnerReplay} isLoading={isReplaying} disabled={!event || !event.eventId}>Replay</Button>
        </div>
    );
};

EventDrawerHeader.propTypes = {
    event: PropTypes.object,
    isReplaying: PropTypes.bool,
    onReplay: PropTypes.func
};

EventDrawerHeader.defaultProps = {
    event: null,
    isReplaying: false,
    onReplay: null
};

const createMapStateToProps = () => {
    const loadingSelector = createLoadingSelector([REPLAY_EVENT]);
    return (state) => ({
        isReplaying: loadingSelector(state)
    });
};
const mapDispatchToProps = (dispatch) => ({
    onReplay: payload => dispatch(replayEvent(payload)),
});

export default connect(createMapStateToProps, mapDispatchToProps)(EventDrawerHeader);