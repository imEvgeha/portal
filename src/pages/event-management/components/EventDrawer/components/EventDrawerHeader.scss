.nexus-c-event-drawer-header__replay-button {
  margin-left: 20px;
}