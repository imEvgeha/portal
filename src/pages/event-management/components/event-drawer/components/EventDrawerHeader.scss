.nexus-c-event-drawer-header__replay-button,
.nexus-c-event-drawer-header__replicate-button {
  margin-left: 20px;
}