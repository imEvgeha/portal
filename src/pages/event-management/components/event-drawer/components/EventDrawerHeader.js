import React from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import Tooltip from '@atlaskit/tooltip';
import NexusDownload from '@vubiquity-nexus/portal-ui/lib/elements/nexus-download/NexusDownload';
import {can} from '@vubiquity-nexus/portal-utils/lib/ability';
import {connect} from 'react-redux';
import {createLoadingSelector} from '../../../../../ui/loading/loadingSelectors';
import {REPLAY_EVENT, REPLICATE_EVENT} from '../../../eventManagementActionTypes';
import {replayEvent, replicateEvent} from '../../../eventManagementActions';
import './EventDrawerHeader.scss';

export const EventDrawerH = ({event, isReplaying, onReplay, isReplicating, onReplicate}) => {
    const {eventId = '', id = ''} = event || {};
    const canReplayAndReplicate = can('create', 'EventManagement') && can('update', 'EventManagement');

    const onInnerReplay = () => {
        const payload = {docId: id};
        onReplay(payload);
    };

    const onInnerReplicate = () => {
        const payload = {docId: id};
        onReplicate(payload);
    };

    return (
        <div className="nexus-c-event-drawer-header">
            <Tooltip content={canReplayAndReplicate ? 'Replay Event' : 'Insufficient permissions'}>
                <Button
                    className="nexus-c-event-drawer-header__replay-button"
                    onClick={onInnerReplay}
                    isLoading={isReplaying}
                    isDisabled={!canReplayAndReplicate || !event || !eventId}
                >
                    Replay
                </Button>
            </Tooltip>

            <Tooltip content={canReplayAndReplicate ? 'Replicate Event' : 'Insufficient permissions'}>
                <Button
                    className="nexus-c-event-drawer-header__replicate-button"
                    onClick={onInnerReplicate}
                    isLoading={isReplicating}
                    isDisabled={!canReplayAndReplicate || !event || !eventId}
                >
                    Replicate
                </Button>
            </Tooltip>
            <NexusDownload
                className="nexus-c-event-drawer-header__download-button"
                data={event}
                filename={`${eventId} - event`}
                isDisabled={!event || !eventId}
            />
        </div>
    );
};

EventDrawerH.propTypes = {
    event: PropTypes.object,
    isReplaying: PropTypes.bool,
    onReplay: PropTypes.func,
    isReplicating: PropTypes.bool,
    onReplicate: PropTypes.func,
};

EventDrawerH.defaultProps = {
    event: null,
    isReplaying: false,
    onReplay: null,
    isReplicating: false,
    onReplicate: null,
};

const createMapStateToProps = () => {
    const replayingLoadingSelector = createLoadingSelector([REPLAY_EVENT]);
    const replicatingLoadingSelector = createLoadingSelector([REPLICATE_EVENT]);
    return state => ({
        isReplaying: replayingLoadingSelector(state),
        isReplicating: replicatingLoadingSelector(state),
    });
};
const mapDispatchToProps = dispatch => ({
    onReplay: payload => dispatch(replayEvent(payload)),
    onReplicate: payload => dispatch(replicateEvent(payload)),
});

export default connect(createMapStateToProps, mapDispatchToProps)(EventDrawerH);
