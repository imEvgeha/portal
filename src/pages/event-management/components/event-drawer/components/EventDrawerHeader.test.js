import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import EventDrawerHeader, {EventDrawerH} from './EventDrawerHeader';

describe('EventDrawerHeader', () => {
    let mockStore;
    let store;
    let onReplayEventMock;
    let onReplicateEventMock;
    let wrapper;
    beforeEach(() => {
        mockStore = configureStore();
        store = mockStore({});
        onReplayEventMock = jest.fn();
        onReplicateEventMock = jest.fn();
        const props = {
            store,
            event: {eventId: '123'},
            onReplay: onReplayEventMock,
            onReplicate: onReplicateEventMock,
        };
        wrapper = shallow(
            <EventDrawerH
                {...props}
            />
        );
    });

    it('should render the component', () => {
        expect(wrapper).not.toBe(null);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should should call onReplay when replay button clicked', () => {
        const replayButton = wrapper.find('.nexus-c-event-drawer-header__replay-button');
        expect(replayButton).toHaveLength(1);
        replayButton.simulate('click');
        expect(onReplayEventMock.mock.calls.length).toEqual(1);
    });

    it('should should call onReplicate when replicate button clicked', () => {
        const replicateButton = wrapper.find('.nexus-c-event-drawer-header__replicate-button');
        expect(replicateButton).toHaveLength(1);
        replicateButton.simulate('click');
        expect(onReplicateEventMock.mock.calls.length).toEqual(1);
    });
});
