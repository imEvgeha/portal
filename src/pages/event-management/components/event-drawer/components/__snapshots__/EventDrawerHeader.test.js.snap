// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDrawerHeader should match snapshot 1`] = `
<div
  className="nexus-c-event-drawer-header"
>
  <WithAnalyticsContext(WithAnalyticsEvents(Button))
    className="nexus-c-event-drawer-header__replay-button"
    isDisabled={false}
    isLoading={false}
    onClick={[Function]}
  >
    Replay
  </WithAnalyticsContext(WithAnalyticsEvents(Button))>
  <WithAnalyticsContext(WithAnalyticsEvents(Button))
    className="nexus-c-event-drawer-header__replicate-button"
    isDisabled={false}
    isLoading={false}
    onClick={[Function]}
  >
    Replicate
  </WithAnalyticsContext(WithAnalyticsEvents(Button))>
  <NexusDownload
    className="nexus-c-event-drawer-header__download-button"
    data={
      Object {
        "eventId": "123",
      }
    }
    filename="123"
    isDisabled={false}
    label="Download"
    mimeType="application/json"
  />
</div>
`;
